packages/drivers/odsp-driver/src/odspDocumentStorageManager.ts

```diff
@@ -340,10 +342,11 @@ export class OdspDocumentStorageService implements IDocumentStorageService {
                     ...this.hostPolicy.snapshotOptions,
                 };
 
-                // No limit on size of snapshot, as otherwise we fail all clients to summarize
-                if (this.hostPolicy.summarizerClient) {
-                    snapshotOptions.mds = undefined;
-                }
+                const maxSnapshotSizeLimit = 25000000; // 25 MB
+                const maxSnapshotFetchTimeout = 30000; // 30 sec
+                // Choose min of host specified limits or driver specified limits.
+                snapshotOptions.mds = snapshotOptions.mds ? Math.min(snapshotOptions.mds, maxSnapshotSizeLimit) : maxSnapshotSizeLimit;
+                snapshotOptions.timeout = snapshotOptions.timeout ? Math.min(snapshotOptions.timeout, maxSnapshotFetchTimeout) : maxSnapshotFetchTimeout;
```

Comment:
This will always override if the host specifies a snapshot limit. Do we generally want to defer to the server for setting the size or are there scenarios where the client might care more? (IE. Mobile might want smaller limits)

Non-block. Just food for thought.

Reply:
I think it should depend on client side because if the client can handle large snapshot or wait for long time, then server should not care.

Reply:
i don't think it will anyway if  snapshotOptions.mds is set, it came from the host via shallow clone, i think moving this above will make that clearer